ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/deneb/helpers/MiscHelpersDeneb.java

```diff
@@ -65,26 +67,43 @@ private static KZG initKZG(final SpecConfigDeneb config) {
   public boolean isDataAvailable(
       final UInt64 slot,
       final Bytes32 beaconBlockRoot,
-      final List<KZGCommitment> kzgCommitments,
+      final List<KZGCommitment> kzgCommitmentsFromBlock,
       final List<BlobSidecar> blobSidecars) {
-    blobSidecars.forEach(
-        blobSidecar -> {
-          checkArgument(
-              slot.equals(blobSidecar.getSlot()),
-              "Blob sidecar slot %s does not match block slot %s",
-              blobSidecar.getSlot(),
-              slot);
-          checkArgument(
-              beaconBlockRoot.equals(blobSidecar.getBlockRoot()),
-              "Blob sidecar block root %s does not match block root %s",
-              blobSidecar.getBlockRoot(),
-              beaconBlockRoot);
-        });
-    final List<Bytes> blobs =
-        blobSidecars.stream().map(BlobSidecar::getBlob).map(Blob::getBytes).toList();
-    final List<KZGProof> proofs = blobSidecars.stream().map(BlobSidecar::getKZGProof).toList();
-
-    return kzg.verifyBlobKzgProofBatch(blobs, kzgCommitments, proofs);
+    checkArgument(
+        blobSidecars.size() == kzgCommitmentsFromBlock.size(),
+        "Blob sidecars count %s does not match KZG commitments count %s",
```

Comment:
maybe `block KZG commitments` to be more clear

Reply:
would this be more helpful to report if we had the actual block slot or something associating it to the block in the message?
It'll be very hard for anyone to understand what this relates to because there's nothing tying it into data we can understand, so from the perspective of a node op, this will be a 'useless message' as it stands.

Reply:
right!